components/base-war/src/main/webapp/templates/macros.vm

```diff
@@ -1418,7 +1418,13 @@ ${Msz}Mb##
     #if ("$!resultPage" == '')
       #set ($resultPage = 'XWiki.LiveTableResults')
     #end
-    #set ($dataurl = $xwiki.getURL($resultPage, 'get', "$escapetool.url($parameters)&$!options.extraParams"))
+    #if ($resultPage.startsWith('/'))
+       #set ($hostname = "$request.getServerName()")
+       #set ($hostport = "$request.getServerPort()")
+       #set ($dataurl = "http://$hostname:$hostport$resultPage?$escapetool.url($parameters)&$!options.extraParams")
```

Comment:
What @sdumitriu is saying is basically:
* If your `$resultsPage` is already a relative url (starting with `/`, e.g. `/rest/entities`), then just use that relative url with the appropriate parameters
* If `$resultsPage` is a document reference, build the relative url of the document and append the parameters.

```
#if ($resultPage.startsWith('/'))
  #set ($hostname = "$request.getServerName()")
  #set ($hostport = "$request.getServerPort()")
  #set ($dataurl = "$resultPage?$escapetool.url($parameters)&$!options.extraParams")
#else
  #set ($dataurl = $xwiki.getURL($resultPage, 'get', "$escapetool.url($parameters)&$!options.extraParams"))
#end
```